packages/next/server/render.tsx

```diff
@@ -274,6 +277,49 @@ function checkRedirectValues(
   }
 }
 
+// Create the wrapper component for a Flight stream.
+function createServerComponentRenderer(
+  OriginalComponent: React.ComponentType,
+  serverComponentManifest: NonNullable<RenderOpts['serverComponentManifest']>
+) {
+  let responseCache: any
+  const ServerComponentWrapper = (props: any) => {
+    let response = responseCache
+    if (!response) {
+      responseCache = response = createFromReadableStream(
+        renderToReadableStream(
+          <OriginalComponent {...props} />,
+          serverComponentManifest
+        )
+      )
+    }
+    return response.readRoot()
+  }
+  const Component = (props: any) => {
+    return (
+      <React.Suspense fallback={null}>
```

Comment:
FYI, Frameworks shouldn't insert suspense boundaries, because the app has no way to customize the fallback/loading state. What do we need this for?

Reply:
This is how it's implemented in Next.js today: https://github.com/vercel/next.js/blob/28a5d7a2e31bb2bfb6a8843bd6542659cac1f2d4/packages/next/build/webpack/loaders/next-middleware-ssr-loader/index.ts#L92-L98

and previously the top-level suspense boundary didn't exist. We can surely get rid of it, but I'm unsure if it will cause hydration mismatch on the client side since we'll need a boundary there.